erpnext/erpnext_integrations/doctype/leaflink_settings/leaflink_settings.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import json
+import pprint
+from frappe.utils import add_days, cint
+import datetime
+import requests
+from frappe.model.document import Document
+import base64
+import hashlib
+import hmac
+from frappe import _
+
+class LeafLinkSettings(Document):
+	pass
+
+@frappe.whitelist()
+def sync_customer_with_leaflink(customer_name):
+	"""
+	Sync a customer with leaflink by hitting Sync with Customer Button.
+	"""
+	leaflink_settings = frappe.get_cached_doc("LeafLink Settings")
+	header = {
+		leaflink_settings.get("api_key"): leaflink_settings.get("api_value")
+	}
+
+	#validate if leaflink is enabled
+	if not leaflink_settings.get("enable_leaflink"):
+		frappe.throw("Please enable LeafLink in LeafLink Settings.")
+
+	#check if customer exists on leaflink
+	try:
+		customers_url = leaflink_settings.get("customers_url")
+	except Exception as e:
+		frappe.throw(_("An Exception: {e} occured while trying to sync customer on LeafLink. Please try again.".format(e)))
+
+	customers_on_leaflink = requests.get(customers_url, headers=header, params={"name": customer_name}).json()
+
+	#create customer record on leaflink if it does not exist
+	if not customers_on_leaflink.get("count"):
+
+		#create new customer data
+		new_customer = {
+			"name": customer_name,
+			"owner": leaflink_settings.get("leaflink_id")
+	 	}
+
+		#post request to create customer in leaflink
+		try:
+			post_response = requests.post(customers_url, headers=header, data=new_customer)
+			if post_response.status_code == 201:
+				frappe.msgprint(_("Customer successfully created on LeafLink!"))
+		except Exception as e:
+			frappe.throw(_("An Exception: {e} occured while trying to sync customer on LeafLink. Please try again.".format(e)))
+
+	else:
+		frappe.msgprint(_("Customer already exists on LeafLink!"))
+
+
+def authenticate_leaflink_request(leaflink_settings, integration_request):
```

Comment:
As mentioned below, no need to create an integration request if the request is not authenticated, please change the function accordingly.

Reply:
I am still for creating an integration request for whenever this endpoint is called - even if the authentication fails, we will have a log of data that was used to make the api call